### PR TITLE
#387: Handle network error

### PR DESCRIPTION
**Pull request by hani-ibrahim**
Working on #387 
img_2549

### CODE REVIEW

**Comment by hani-ibrahim** [comment]
👍

File: AtlasUI/AtlasUI/AtlasUIErrors.swift [hunk 1]
```diff
@@ -52,12 +52,19 @@ extension UserPresentable {
 
 extension AtlasAPIError: UserPresentable {
 
+    func shouldDisplayGeneralMessage() -> Bool {
+        switch self {
+        case let .nsURLError(_, details): return details == nil
```
Comment (bonkey) [inline AtlasUI/AtlasUI/AtlasUIErrors.swift]: Is it reasonable? Should details of NSURLError always go to the user? They're usually pretty reasonable, but I would double check it...

**Comment by bonkey** [comment]
👍

File: AtlasUI/AtlasUI/AtlasUIErrors.swift (hunk 1 above)
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/AtlasUIErrors.swift]: I think yes as they already have localized description to be displayed to the end user. Any apple app that have a connection problem display these errors.

However, they are pretty many errors 😖 
https://developer.apple.com/reference/foundation/1508628-url_loading_system_error_codes

**Comment by bonkey** [comment]
:+

**Pull request merged**